packages/common/src/store/pages/chat/slice.ts

```diff
@@ -175,24 +188,27 @@ const slice = createSlice({
         chatId,
         response: { data, summary }
       } = action.payload
-      chatsAdapter.updateOne(state.chats, {
-        id: chatId,
-        changes: {
-          messagesStatus: Status.SUCCESS,
-          messagesSummary: summary
-        }
-      })
-      chatsAdapter.updateOne(state.chats, {
-        id: chatId,
-        changes: { messagesStatus: Status.SUCCESS, messagesSummary: summary }
-      })
+      if (!summary) {
+        console.error('fetchMoreMessagesSucceeded: no summary')
+        return
+      }
+      const existingSummary = state.chats.entities[chatId]?.messagesSummary
+      if (summary.next_count > (existingSummary?.next_count ?? -1)) {
```

Comment:
I think we really want to take the max of the `next_cursor` and the min of the `prev_cursor` (or vice versa? can't remember which way they go)?

Reply:
Chatted with @rickyrombo on slack, his way is the way to go. "the summary we get back is for that query but I want the store to have the summary for all the stuff itself."

Reply:
Turns out we want min of both i believe.